src/Helper/Asset.php

```diff
@@ -7,31 +7,49 @@
 use Laminas\View\Exception;
 
 use function array_key_exists;
+use function sprintf;
 
 /**
  * View helper plugin to fetch asset from resource map.
+ *
+ * @final
  */
 class Asset extends AbstractHelper
 {
-    /** @var array */
+    use DeprecatedAbstractHelperHierarchyTrait;
+
+    /** @var array<non-empty-string, non-empty-string> */
     protected $resourceMap = [];
 
     /**
-     * @param string $asset
-     * @return string
+     * @param array<non-empty-string, non-empty-string> $resourceMap
+     */
+    public function __construct(array $resourceMap = [])
+    {
+        $this->resourceMap = $resourceMap;
+    }
+
+    /**
+     * @param non-empty-string $asset
+     * @return non-empty-string
      * @throws Exception\InvalidArgumentException
      */
     public function __invoke($asset)
     {
         if (! array_key_exists($asset, $this->resourceMap)) {
-            throw new Exception\InvalidArgumentException('Asset is not defined.');
+            throw new Exception\InvalidArgumentException(sprintf(
+                'The asset with the name "%s" has not been defined.',
+                $asset
+            ));
```

Comment:
Is this a potential security issue? No attempt is made to escape the key, so if the user is showing exception messages and allowing untrusted input for the asset key, this could cause issues, or is it 'not our responsibility' if the user chooses to show exception messages in production?

Reply:
> so if the user is showing exception messages

This would be the security issue: showing infrastructural exception messages is never supposed to happen, so `Asset.php` is not responsible for such a fault.

Reply:
Thanks. I thought so… but it was worth making sure!